### PR TITLE
Fix broken rules with Prometheus v3

### PR DESCRIPTION
**Pull request by leonnicolas**
With Prometheus v3 [histograms are normalized](https://prometheus.io/docs/prometheus/latest/migration/#le-and-quantile-label-values) and this breaks rules that
select on the `le` label.
This PR updates the label selectors to use the new normalized value.

Before:
```
{le="1"}
```
Now with Prometheus v3:
```
{le="1.0"}
```

Fixes issue: https://github.com/kubernetes-monitoring/kubernetes-mixin/issues/1005

Signed-off-by: leonnicolas <leonloechner@gmx.de>

### CODE REVIEW

**Comment by leonnicolas** [comment]
> Does it / will it work Prometheus 2?

No, If we want to support both, I guess we could do something like
```
metrics{ls~="1|1.0"}
```
Would be fine with me. I have no idea how much this could impact load on Prometheus

**Comment by leonnicolas** [comment]
How would a feature flag work/look like? This gave me the idea to fix this downstream in kube-prometheus with this [PR](https://github.com/prometheus-operator/kube-prometheus/pull/2575).

**Comment by skl** [comment]
@povilasv @leonnicolas I was going to reply here but thought it would be easier to create a PR instead:

- #1008 

TL;DR

- No performance issue observed using regex matcher on these rules
- Suggest we go the regex route to support both v2/v3 prometheus as well as the issues reported in #965

Let me know what you think.
